app/lib/actions/katello/upstream_subscriptions/remove_entitlement.rb

```diff
@@ -6,10 +6,13 @@ class RemoveEntitlement < Actions::Base
 
         input_format do
           param :entitlement_id
+          param :sub_name
         end
 
         def run
           output[:response] = ::Katello::Resources::Candlepin::UpstreamConsumer.remove_entitlement(input[:entitlement_id])
+        rescue RestClient::NotFound
```

Comment:
Can you give remove_entitlement the same treatment and raise a custom error which masks RestClient::NotFound?

Reply:
Would be nice to have a test for the behavior change here (adding a custom message into the output response)

Reply:
Sounds good, my latest commit should address these.